ecommerce/models.py

```diff
@@ -0,0 +1,33 @@
+from django.db import models
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from mitol.common.models import TimestampedModel
+import reversion
+
+@reversion.register(
+    exclude=("content_type", "object_id", "created_on", "updated_on")
+)
+class Product(TimestampedModel):
+    """
+    Representation of a purchasable product. There is a GenericForeignKey to a 
+    Course Run or Program Run.
+    """
+
+    valid_purchasable_objects=models.Q(app_label='courses', model='courserun') | models.Q(app_label='courses', model='programrun')
+    content_type=models.ForeignKey(ContentType, on_delete=models.CASCADE, limit_choices_to=valid_purchasable_objects)
```

Comment:
Could you change the value passed to `limit_choices_to` to be a function returning the `Q` fragment instead? That has the benefit that if we ever modify this query it doesn't trigger a new migration to be generated since Django will just serialize the function name into the migration instead of the query itself.